src/main/connect/connectionManager.ts

```diff
@@ -748,19 +782,113 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
-        let policyMessage: string = '';
+        let client: JfrogClient = this.createJfrogClient();
+        let scanService: XrayScanClient = client.xray().scan();
+        let technology: string[] | undefined;
+        if (msi && msi !== '' && packageType) {
+            scanService = client.xsc().scan();
+            technology = [packageType];
+        }
+        this._logManager.logMessage(this.getScanLog(graphRequest.component_id, project, watches, msi, packageType), 'DEBUG');
+        return await scanService.graph(graphRequest, progress, checkCanceled, project, watches, msi, technology);
+    }
+
+    private getScanLog(componentId: string, project: string, watches: string[], msi?: string, packageType?: string): string {
+        let service: string = 'Xray';
+        let message: string = '';
+        if (msi && msi !== '') {
+            service = 'XSC';
+            message += ` Scan MSI: ${msi}`;
+            if (packageType) {
+                message += ` Scan Technology: ${packageType}`;
+            }
+        }
         if (watches.length > 0) {
-            policyMessage += ` Using Watches: [${watches.join(', ')}]`;
+            message += ` Using Watches: [${watches.join(', ')}]`;
         } else if (project && project !== '') {
-            policyMessage += ` Using Project key: ${project}`;
+            message += ` Using Project key: ${project}`;
         }
-        this._logManager.logMessage('Sending dependency graph "' + graphRequest.component_id + '" to Xray for analyzing.' + policyMessage, 'DEBUG');
-        return await this.createJfrogClient()
-            .xray()
-            .scan()
-            .graph(graphRequest, progress, checkCanceled, project, watches);
+        return `Sending dependency graph "` + componentId + `" to ` + service + ` for analyzing.` + message;
+    }
+
+    public async startScan(request: StartScanRequest): Promise<ScanEvent> {
+        if (this._xscUrl === '') {
+            this._logManager.logMessage('Xsc is not configured. Skipping analytics...', 'DEBUG');
+            return {} as ScanEvent;
+        }
+        return this.createJfrogClient()
+            .xsc()
+            .event()
+            .startScan(request);
+    }
+
+    public async endScan(event: ScanEvent): Promise<void> {
+        if (this._xscUrl === '') {
+            return;
+        }
+        this.createJfrogClient()
+            .xsc()
+            .event()
+            .endScan(event);
+    }
+
+    public async logWithAnalytics(message: string, level: LogLevel): Promise<void> {
+        if (this._xscUrl === '') {
+            return;
+        }
+        this.createJfrogClient()
+            .xsc()
+            .event()
+            .log({
+                message: message,
+                source: 'jfrog-vscode-extension',
+                log_level: this.toXscLogLevel(level)
+            });
+    }
+
+    public async logErrorWithAnalytics(err: Error, shouldToast: boolean = false): Promise<void> {
```

Comment:
I wouldn't call it `logErrorWithAnalytics`, I'd call something like "logError" - since this is our new default, and the handle inside would be something like...
`await this.reportLog(`${err.name}: ${err.message}\n${err.stack}`, 'ERR');`
after the validation that we `shouldReport`